packages/sanity/src/field/diff/resolve/resolveDiffComponent.ts

```diff
@@ -12,7 +12,7 @@ export function resolveDiffComponent<D extends Diff = any>(
 
   while (itType) {
     const resolved =
-      (itType as any).diffComponent ||
+      (itType as any)?.components?.diff ||
```

Comment:
Can we remove the `as any` now?

```suggestion
      itType?.components?.diff ||
```